mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1412,10 +1472,13 @@ where
         amount: big_decimal_from_sat(fee_amount as i64, coin.as_ref().decimals),
     };
     let my_address = coin.my_address().map_to_mm(WithdrawError::InternalError)?;
-    let to_address = coin.display_address(&to).map_to_mm(WithdrawError::InternalError)?;
+    let tx_hex = match conf.address_format {
+        UtxoAddressFormat::Segwit => serialize_with_flags(&signed, SERIALIZE_TRANSACTION_WITNESS).into(),
```

Comment:
https://github.com/KomodoPlatform/atomicDEX-API/blob/mm2.1-segwit-withdraw/mm2src/mm2_bitcoin/chain/src/transaction.rs#L209
Can't generate permalink for this function because file is too big to be viewed in github so I wrote it here 

```
pub fn hash(&self) -> H256 {
        let serialized = &serialize(self);
        match self.tx_hash_algo {
            TxHashAlgo::DSHA256 => dhash256(&serialized),
            TxHashAlgo::SHA256 => sha256(&serialized),
        }
    }
```

So in generating the `tx_hash` the transaction is serialized without the witness data to prevent transaction malleability https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#Transaction_ID but I will make a different function to serialize the transaction without witness if it has one to use it in the hash function and will use only one serialize function in the withdraw method.

Reply:
I see, let's keep it as is then, thanks for the info :slightly_smiling_face: